docs/configuration.md

```diff
@@ -160,14 +160,20 @@ Here are the catalog properties related to locking. They are used by some catalo
 
 The following properties from the Hadoop configuration are used by the Hive Metastore connector.
 
-| Property                              | Default          | Description                                                                        |
-| ------------------------------------- | ---------------- | ---------------------------------------------------------------------------------- |
-| iceberg.hive.client-pool-size         | 5                | The size of the Hive client pool when tracking tables in HMS                       |
-| iceberg.hive.lock-timeout-ms          | 180000 (3 min)   | Maximum time in milliseconds to acquire a lock                                     |
-| iceberg.hive.lock-check-min-wait-ms   | 50               | Minimum time in milliseconds to check back on the status of lock acquisition       |
-| iceberg.hive.lock-check-max-wait-ms   | 5000             | Maximum time in milliseconds to check back on the status of lock acquisition       |
-
-Note: `iceberg.hive.lock-check-max-wait-ms` should be less than the [transaction timeout](https://cwiki.apache.org/confluence/display/Hive/Configuration+Properties#ConfigurationProperties-hive.txn.timeout) 
+| Property                                  | Default         | Description                                                                  |
+|-------------------------------------------|-----------------|------------------------------------------------------------------------------|
+| iceberg.hive.client-pool-size             | 5               | The size of the Hive client pool when tracking tables in HMS                 |
+| iceberg.hive.lock-timeout-ms              | 180000 (3 min)  | Maximum time in milliseconds to acquire a lock                               |
```

Comment:
I'm a little confused on the difference between the lock timeout, lock check and lock request.

They all seem to have very similar descriptions. Could you clarify which is which?

Reply:
Tried to rename and add some description.
Would this be better?

Reply:
I think that is a bit clearer for me.

Reply:
If you have even better suggestion, I am open to update, so feel free to comment 😄

Reply:
I must have missed the earlier discussion, but should 'iceberg.hive.lock-timeout-ms' be 
'iceberg.hive.lock-check-timeout' as its used with iceberg.hive.lock-check-min-wait-ms.min-wait-ms and iceberg.hive.lock-check-min-wait-ms.max-wait-ms?

I guess the two phases to me would be  :  
1. Lock Creation request
2. Lock Check (check for successful acquisition of lock)

Reply:
`iceberg.hive.lock-timeout-ms` is an already existing configuration. Is there an existing way to seamlessly deprecate configuration keys in favor of a new one? I did not want to break existing uses of this configuration key.

Reply:
I see, just saw its from unrelated changelist.  But it still doesnt fit very well with the two phases (creation/acquisition) you put in the first part of the doc, should we just say simply:


The HMS table locking is a 2-step process:
1. Lock create: Create lock in HMS and queue for acquisition
2. Lock check: Check if lock successfully acquired

This way, they are more relevant to the property name.  Also wasnt sure about putting the details of 'should wait until every previously created Lock is released', is this the Hive internals?  As I feel it will just confuse more.

Reply:
Done